test/functional/apps/discover/observability/logs/_get_pagination_config.ts

```diff
@@ -0,0 +1,125 @@
+    to: '2024-06-10T16:30:00.000Z',
+  };
+
+  const currentTimeFrame = {
+    from: '2015-09-20T01:00:00.000Z',
+    to: '2015-09-24T16:30:00.000Z',
+  };
+
+  describe('extension getPaginationConfig', () => {
+    before(async () => {
+      // To load more than 500 records
+      await esArchiver.loadIfNeeded('test/functional/fixtures/es_archiver/logstash_functional');
+      await kibanaServer.uiSettings.update({
+        'timepicker:timeDefaults': `{ "from": "${currentTimeFrame.from}", "to": "${currentTimeFrame.to}"}`,
+      });
+    });
+
+    after(async () => {
+      await esArchiver.unload('test/functional/fixtures/es_archiver/logstash_functional');
+      await kibanaServer.uiSettings.update({
+        'timepicker:timeDefaults': `{ "from": "${resetTimeFrame.from}", "to": "${resetTimeFrame.to}"}`,
+      });
+    });
+
+    describe('ES|QL mode', () => {
+      it('should render without pagination using a single page', async () => {
+        const limit = 200;
+        const state = kbnRison.encode({
+          dataSource: { type: 'esql' },
+          query: { esql: `from logstash* | sort @timestamp desc | limit ${limit}` },
+        });
+        await PageObjects.common.navigateToActualUrl('discover', `?_a=${state}`, {
+          ensureCurrentUrl: false,
+        });
+        await PageObjects.discover.waitUntilSearchingHasFinished();
+
+        await dataGrid.scrollTo(300);
+
+        await PageObjects.discover.waitUntilSearchingHasFinished();
+        // In ESQL Mode, pagination is disabled
+        await testSubjects.missingOrFail('tablePaginationPopoverButton');
+        await testSubjects.missingOrFail('pagination-button-previous');
+        await testSubjects.missingOrFail('pagination-button-next');
+      });
+    });
+
+    describe('data view mode', () => {
+      it('should render default pagination with page numbers', async () => {
+        await PageObjects.common.navigateToActualUrl('discover', undefined, {
+          ensureCurrentUrl: false,
+        });
+        await dataViews.createFromSearchBar({
+          name: 'lo', // Must be anything but log/logs, since pagination is disabled for log sources
+          adHoc: true,
+          hasTimeField: true,
+        });
+
+        await PageObjects.discover.waitUntilSearchingHasFinished();
+
+        await testSubjects.existOrFail('tablePaginationPopoverButton');
+        await testSubjects.existOrFail('pagination-button-previous');
+        await testSubjects.existOrFail('pagination-button-next');
+        await dataGrid.checkCurrentRowsPerPageToBe(100);
+      });
+
+      it('should render single page pagination without page numbers', async () => {
+        const defaultPageLimit = 500;
+        await PageObjects.common.navigateToActualUrl('discover', undefined, {
+          ensureCurrentUrl: false,
+        });
+        await dataViews.createFromSearchBar({
+          name: 'logs', // Must be anything but log/logs, since pagination is disabled for log sources
```

Comment:
```suggestion
          name: 'logs',
```
I think this one is more like "not anything but logs" 😁